content/news/XXXX-XX-XX-dvs-internals-pt2.md

```diff
@@ -0,0 +1,138 @@
+If you're a programmer, you might say: "We can use sentinel values to solve this problem."
+A [sentinel value](https://en.wikipedia.org/wiki/Sentinel_value) is a special value that is distinct from all other legal values.
+Such a value could be used to indicate "A new position value starts here".
+
+After a seek, we have to keep reading bits until we encounter the sentinel value, then we can start reading the actual position bits.
+In fact, that's exactly what the early [Final Scratch DVS system](https://en.wikipedia.org/wiki/Final_Scratch#Vinyl/CD_time_code) did.
+Each position was prefixed by `0001`.
+Of course, that bit sequence must not be part of a position bit sequence, but that can be solved by using a slightly more complicated encoding.
+
+#### Example: Final Scratch
+
+The Final Scratch timecode uses a 4 bit sentinel value followed by 16 bit of position information.
+Usually, 16 bits can represent 2<sup>16</sup> = 65536 distinct values.
+But we need to filter out all values that contain the sequence `0001`, because that would look just like the sentinel value and confuse the software.
+If we do that, only 27663 distinct values remain.
+
+That is not enough to represent all positions with sufficient resolution, which is why they had to work around that problem by using both the left and right channels to encode information (in contrast to Serato, which only uses the left channel).
+On each channel, a position starts with the sentinel value `0001`, then 16 bits of position information (different on each channel), so 40 bits per position in total.
+Because these bits are split across both channels, this is equivalent to 20 bits of information of a single-channel bitstream in terms of latency.
+
+The downside is that you can't use the right channel for error detection.
+If the bits on both channels are not the same anymore, you cannot compare the bits from both channels to rule out read errors caused by dust on the record.
+
+But the bigger issue is that using sentinel values may still require a lot more bits to be read after a seek before you can detect the position reliably.
+In the worst case, the DJ seeks to a position where the first bit of the sentinel value is missing.
+Then we have to read the remaining bits of the sentinel value, then all bits of the full position, then next sentinel value and then all bits of the next full position value before we know what position we're at.
+Needing so many bits to reliably detect a position is bad for latency and might make the system feel "sluggish".
+
+
+### Using an LFSR
+
+It would be great if we could reduce the number of bits that we need to read to detect the position reliably - even in the worst case.
+One way to do this, is by using a Linear Feedback Shift Register (LFSR).
+
+LFSRs are commonly used, for example as pseudo random number generators for stream ciphers.
+Someone at Serato had the interesting idea to use them for encoding positions on their timecode vinyl instead.
```

Comment:
I would move this eye scathing sentence above and introduce LFSR afterwards. I expect that this will attrackt the reader better.